modules/node-shared/src/test/scala/org/tessellation/node/shared/domain/seedlist/SnapshotOrdinalTimelineSuite.scala

```diff
@@ -0,0 +1,91 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import org.tessellation.node.shared.domain.seedlist.generators.terminatedOrdinalRangeGen
+import org.tessellation.node.shared.domain.seedlist.snapshotOrdinalTimeline.{SnapshotOrdinalTimeline, decode}
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object SnapshotOrdinalTimelineSuite extends SimpleIOSuite with Checkers {
+
+  test("parse input") {
+    val gen = for {
+      count <- Gen.chooseNum(1, 5)
+      b <- Gen.listOfN(count, terminatedOrdinalRangeGen)
+      c = b.tail.foldLeft(b.head._3) {
+        case (acc, (_, _, str)) =>
+          acc + ";" + str
+      }
+      ranges = b.map { case (a, b, _) => (a, b) }.map {
+        case (start, end) =>
+          Range(start, end)
+      }
+      entries = ranges
+        .sortBy(_.start.value)
+        .foldLeft(List.empty[Range]) {
+          case (prev :: tail, curr) if curr.start - prev.end <= 1 =>
+            val merged = Range(
+              prev.start,
+              NonNegLong.unsafeFrom(
+                Math.max(prev.end, curr.end)
+              )
+            )
+
+            merged :: tail
+          case (acc, curr) =>
+            curr :: acc
+        }
+        .reverse
+        .toNel
+        .get
+    } yield (entries, c)
+
+    forall(gen) {
+      case (entries, str) =>
+        val actual = decode(str).toEither
+        val expected = SnapshotOrdinalTimeline(entries).asRight[Throwable]
+
+        expect.same(expected, actual)
```

Comment:
I didn't see that before, but please use `expect.eql` as it's a safer way than `expect.same`. Take a look how `.same` fallbacks to universal equality if there is no `Eq`

Reply:
Changed all `expect.same` to `expect.eql` and added implicits as needed.

Reply:
@cngo-github no implicits are needed like you did. please make sure the original class has `Eq` derived and that's it

Reply:
Added `@derive(eqv, show)` to the errors and case classes as appropriate.